TestClasses/ReadWriteTest.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+import sys
+import yaml
+from BaseTest import BaseTest
+from PerformanceTest import PerformanceTest
+import subprocess
+import os
+import time
+import numpy as np
+from zlib import adler32
+
+class ReadWriteTest(BaseTest, PerformanceTest):
+    def __init__(self, configFile=None, singleEndPoint=None, singleProt:list=None):
+        super().__init__()
+        self.groupedOutput = []
+        self.checksums = []
+
+        if configFile is not None:
```

Comment:
configfile parsing can be put in baseTest init. You can then call super().__init__(configFile) here. the values can be read into a 'config' variable. This means calling self.config.value in the function, but makes it more legible and can be reused for all classes

Reply:
you can also have a single config file, with each derived class using only the values needed for it from that struct